master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -64,17 +66,20 @@ public abstract class AbstractMetaManager implements IMetadataHandler {
   public final Map<String, Set<Integer>> registeredAppAndShuffles =
       JavaUtils.newConcurrentHashMap();
   public final Set<String> hostnameSet = ConcurrentHashMap.newKeySet();
-  public final Set<WorkerInfo> workers = ConcurrentHashMap.newKeySet();
+  private final Map<String, WorkerInfo> workersMap = JavaUtils.newConcurrentHashMap();
 
   public final ConcurrentHashMap<WorkerInfo, Long> lostWorkers = JavaUtils.newConcurrentHashMap();
```

Comment:
I have seen your changes. I think these four variables can be changes to use String instead of WorkerInfo to be the key. But there is no need to do it in this PR.

Reply:
sure

Reply:
Have followed up them in this PR

Reply:
It is better to address all of these collections in one PR.

Otherwise, some of the collections check the workerId and others check the workerInfo, it is difficult to maintain the code.